### PR TITLE
Add support for CircleCI

### PR DESCRIPTION
**Pull request by aehyvari**
Here's an idea how we could do CI in CircleCI.

### CODE REVIEW

**Comment by aehyvari** [comment]
It looks like we have to merge this pull request to master before I can enable the branch conditions for the jobs defined in circleci yaml.

**Comment by blishko** [comment]
OK, let's merge it and see.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            - run:
+                name: Clean build directory
+                command: |
+                    rm -rf build
+                    mkdir build
+            - run:
+                name: Install clang
+                command: |
+                    sudo apt-get install clang
+                    sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 60
```
Comment (blishko) [inline .circleci/config.yml:45]: This changes the C++ compiler to point to clang so that the next commands will compile with clang?

**Review by blishko — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (aehyvari) [reply .circleci/config.yml:45]: Correct.  debian uses this update-alternatives system to support several compilers.

**Pull request merged**